server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/text/DeleteSurroundingText.java

```diff
@@ -3,78 +3,62 @@
-
-        final BaseInputConnection baseInputConnection = (BaseInputConnection) inputConnection;
-
-        final Editable editable = baseInputConnection.getEditable();
-
-        if (editable == null) {
-            return Result.failedResult("Unable to set selection, not editable");
+            throw new IllegalArgumentException(USAGE);
         }
 
-        final int argBeforeLength, argAfterLength;
-
         try {
             argBeforeLength = Integer.parseInt(args[0]);
             argAfterLength = Integer.parseInt(args[1]);
         } catch (NumberFormatException e) {
-            return Result.failedResult(USAGE);
+            throw new IllegalArgumentException(USAGE);
         }
+    }
+
+    @Override
+    protected String getNoFocusedViewMessage() {
+        return "Unable to delete surrounding text, no element has focus";
+    }
 
-        InstrumentationBackend.solo.runOnMainSync(new Runnable() {
-            @Override
-            public void run() {
-                // Find length of non-formatted text
-                int textLength = InfoMethodUtil.getEditableTextLength(editable);
-                int beforeLength, afterLength;
+    @Override
+    protected Result executeOnUIThread(final View servedView, final InputConnection inputConnection) {
+        final Editable editable = InfoMethodUtil.getEditable(servedView);
 
-                if (argBeforeLength < 0) {
-                    beforeLength = textLength + argBeforeLength + 1;
-                } else {
-                    beforeLength = argBeforeLength;
-                }
+        // Find length of non-formatted text
+        int textLength = InfoMethodUtil.getEditableTextLength(editable);
+        int beforeLength, afterLength;
 
-                if (argAfterLength < 0) {
-                    afterLength = textLength + argAfterLength + 1;
-                } else {
-                    afterLength = argAfterLength;
-                }
+        if (argBeforeLength < 0) {
+            beforeLength = textLength + argBeforeLength + 1;
+        } else {
+            beforeLength = argBeforeLength;
+        }
 
-                if (Build.VERSION.SDK_INT >= 9) {
-                    int start = Selection.getSelectionStart(editable);
-                    int end = Selection.getSelectionEnd(editable);
-                    baseInputConnection.setComposingRegion(start, end);
-                }
+        if (argAfterLength < 0) {
+            afterLength = textLength + argAfterLength + 1;
+        } else {
+            afterLength = argAfterLength;
+        }
+
+        if (Build.VERSION.SDK_INT >= 9) {
+            int start = Selection.getSelectionStart(editable);
+            int end = Selection.getSelectionEnd(editable);
+            inputConnection.setComposingRegion(start, end);
+        }
 
-                baseInputConnection.deleteSurroundingText(beforeLength, afterLength);
-            }
-        });
+        inputConnection.deleteSurroundingText(beforeLength, afterLength);
```

Comment:
Here's a good reason why we should use `editable.clear()` instead of `inputConnection.deleteSurroundingText()`:

If the user has their own implementation of an `InputConnection`, they can leave `deleteSurroundingText()` as a no-op and it will have no effect. I believe in this case calling `editable.clear()` will be safer.